master/pkg/schemas/expconf/legacy.go

```diff
@@ -0,0 +1,175 @@
+package expconf
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/pkg/schemas"
+)
+
+// LegacyConfig represents a config that was once valid but might no longer be shimmable to the
+// current version of ExperimentConfig.  Fortunately, the parts of the system that need to read
+// configs that are this old are either 1) only concerned with the raw config for rendering it to
+// the user, or 2) only need very specific parts of the old config, so to reduce our own burden we
+// only expose very specific data from it.
+//
+// LegacyConfig can be used to deal with configs that contain some EOL components, like searchers
+// that have been removed.
+type LegacyConfig struct {
+	checkpointStorage CheckpointStorageConfig
+	bindMounts        BindMountsConfig
+	envvars           EnvironmentVariablesMap
+}
+
+// CheckpointStorage returns a current CheckpointStorage from a LegacyConfig.
+func (h LegacyConfig) CheckpointStorage() CheckpointStorageConfig {
+	return h.checkpointStorage
+}
+
+// BindMounts returns a current BindMountsConfig from a LegacyConfig.
+func (h LegacyConfig) BindMounts() BindMountsConfig {
+	return h.bindMounts
+}
+
+// EnvironmentVariables returns a current EnvironmentVariables from a LegacyConfig.
+func (h LegacyConfig) EnvironmentVariables() EnvironmentVariablesMap {
+	return h.envvars
+}
+
+func getCheckpointStorage(raw map[string]interface{}) (CheckpointStorageConfig, error) {
+	cs := CheckpointStorageConfig{}
+
+	csOnly := raw["checkpoint_storage"]
+
+	csByts, err := json.Marshal(csOnly)
+	if err != nil {
+		return cs, errors.Wrap(err, "unable to remarshal config bytes as json")
+	}
+
+	// Read the checkpoint storage.
+	if err = schemas.SaneBytes(&cs, csByts); err != nil {
+		return cs, errors.Wrap(err, "legacy checkpoint storage does not pass sanity checks")
+	}
+	if err = json.Unmarshal(csByts, &cs); err != nil {
+		return cs, errors.Wrap(err, "unable to unmarshal checkpoint storage bytes")
+	}
+
+	// Fill defaults (should be a no-op).
+	cs = schemas.WithDefaults(cs).(CheckpointStorageConfig)
+
+	// Validate fully before passing anything out.
+	if err = schemas.IsComplete(cs); err != nil {
+		return cs, errors.Wrap(err, "legacy checkpoint storage is incomplete")
+	}
+
+	return cs, nil
+}
+
+func getBindMounts(raw map[string]interface{}) (BindMountsConfig, error) {
+	bm := BindMountsConfig{}
```

Comment:
```suggestion
	var bm BindMountsConfig
```
only because it's really confusing to read a slice init'd that way

Reply:
Sure.  Though I would like to point out that these initialization techniques are materially different.

Reply:
> Though I would like to point out that these initialization techniques are materially different.

Wait, how?

Reply:
also i dont actually care, you can leave it

Reply:
> Wait, how?

The `var` syntax creates a `nil` map, the `{}` syntax creates a non-nil map.  Which is typically what I expect `{}` to do.